### PR TITLE
Crunchyroll missing thumbnail fix

### PR DESCRIPTION
**Pull request by Burve**
When thumbnail is missing returning json skips it instead of the NoneType error

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

DESCRIPTION
When Crunchyroll is missing thumbnail image processing code, here throws NoneType error and does not process the episode to the end.
Fixes #
During processing, if any issues come from a thumbnail, the result got ignored, and thumbnail data is missing from yt-dlp resulting JSON data.

### CODE REVIEW

**Comment by HobbyistDev** [comment]
Fixes #4399

**Comment by pukkandan** [comment]
test this with videos that have thumbnail

**Comment by Burve** [comment]
Tested works fine.
For example, https://www.crunchyroll.com/parallel-world-pharmacy episode 3 returns https://beta.crunchyroll.com/imgsrv/display/thumbnail/1920x1080/catalog/crunchyroll/75e03acdd38124b90e27f641b9353439.jpeg

**Comment by pukkandan** [comment]
ah, it seems I misunderstood the code a bit. But while looking into this, I found similar issues for formats, subtitles etc. Pls check my changes haven't broken anything

**Comment by Burve** [comment]
Hmm, whatever you changed in that two commits broke subtitle downloading. I am sure subtitles was fine before that changes:

```
[debug] Command-line config: ['-u', 'PRIVATE', '-p', 'PRIVATE', '-j', '-vU', 'https://beta.crunchyroll.com/watch/G7PU4ZDEZ/a-reincarnated-pharmacologist-and-a-parallel-world']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.07.18 [135f05e]
[debug] Python 3.10.1 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-107501-g2d924b3a63-20220723 (setts), ffprobe N-107501-g2d924b3a63-20220723
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.0
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[crunchyroll:beta] Getting session id
[crunchyroll:beta] Logging in
[debug] [crunchyroll:beta] Extracting URL: https://beta.crunchyroll.com/watch/G7PU4ZDEZ/a-reincarnated-pharmacologist-and-a-parallel-world
[crunchyroll:beta] Retrieving main page
[crunchyroll:beta] Authenticating with cookie
[crunchyroll:beta] Retrieving signed policy
[crunchyroll:beta] a-reincarnated-pharmacologist-and-a-parallel-world: Retrieving episode metadata
[crunchyroll:beta] a-reincarnated-pharmacologist-and-a-parallel-world: Retrieving stream info
[crunchyroll:beta] a-reincarnated-pharmacologist-and-a-parallel-world: Downloading adaptive_hls HLS manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: 'str' object has no attribute 'get'
Traceback (most recent call last):
  File "C:\Users\AlexI\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1431, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\AlexI\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1521, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "C:\Users\AlexI\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 1578, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "C:\Users\AlexI\AppData\Local\Programs\Python\Python310\lib\site-packages\yt_dlp\YoutubeDL.py", line 2443, in process_video_result
    if subtitle_format.get('url'):
AttributeError: 'str' object has no attribute 'get'
```

File: yt_dlp/extractor/crunchyroll.py [hunk 1]
```diff
@@ -872,19 +852,29 @@ def _real_extract(self, url):
...
+            'thumbnails': [{
+                'url': thumb.get('source'),
+                'width': thumb.get('width'),
+                'height': thumb.get('height'),
+            } for thumb in traverse_obj(episode_response, ('images', 'thumbnail', ..., ...)) or []],
+            'subtitles': {
+                lang: {
+                    'url': subtitle_data.get('url'),
+                    'ext': subtitle_data.get('format')
+                } for lang, subtitle_data in get_streams('subtitles')
```
Comment (pukkandan) [inline yt_dlp/extractor/crunchyroll.py:876]:
```suggestion
                lang: [{
                    'url': subtitle_data.get('url'),
                    'ext': subtitle_data.get('format')
                }] for lang, subtitle_data in get_streams('subtitles')
```

**Comment by pukkandan** [comment]
should be fixed now

**Comment by Burve** [comment]
I tested with the link that generated subtitle error, and it was fine.
I tested with the link that initially had no thumbnail, and it was fine.

**Pull request merged**